crates/vault-registry/src/lib.rs

```diff
@@ -1514,7 +1512,7 @@ impl<T: Config> Pallet<T> {
 
     /// Private getters and setters
 
-    fn get_rich_vault_from_id(vault_id: &T::AccountId) -> Result<RichVault<T>, DispatchError> {
+    pub fn get_rich_vault_from_id(vault_id: &T::AccountId) -> Result<RichVault<T>, DispatchError> {
```

Comment:
This should remain private as it exposes *a lot* of the underlying functionality.